server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -733,6 +734,7 @@ private static boolean awaitTermination(
 
     @Override
     public void close() {
+        DeprecationLogger.removeThreadContext(threadContext);
```

Comment:
I think that is a good decisions but if you do this lets make sure we own the adding here too. I think we should move the part that we have in Node.java into the `ThreadPool` constructor. That way we own the entire lifecycle?